src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -58,28 +63,40 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
         }
 
         public void ExecuteAnalysis(string path, string charset, IEnumerable<AnalysisLanguage> detectedLanguages,
-            IIssueConsumer consumer, ProjectItem projectItem, IAnalyzerOptions analyzerOptions,
+            IIssueConsumer consumer, IAnalyzerOptions analyzerOptions,
             CancellationToken cancellationToken)
         {
+            var projectItem = dte?.Solution?.FindProjectItem(path);
+            if (projectItem == null)
+            {
+                return;
+            }
+
             Debug.Assert(IsAnalysisSupported(detectedLanguages));
 
-            var request = CFamilyHelper.CreateRequest(logger, projectItem, path, cFamilyRulesConfigProvider, analyzerOptions);
+            var request = CreateRequest(logger, projectItem, path, cFamilyRulesConfigProvider, analyzerOptions);
             if (request == null)
             {
                 return;
             }
 
+            TriggerAnalysis(request, consumer, cancellationToken);
+        }
+
+        protected /* for testing */ virtual Request CreateRequest(ILogger logger, ProjectItem projectItem, string absoluteFilePath, ICFamilyRulesConfigProvider cFamilyRulesConfigProvider, IAnalyzerOptions analyzerOptions) =>
+            CFamilyHelper.CreateRequest(logger, projectItem, absoluteFilePath, cFamilyRulesConfigProvider, analyzerOptions);
+
+        protected /* for testing */ virtual void TriggerAnalysis(Request request, IIssueConsumer consumer, CancellationToken cancellationToken) =>
```

Comment:
This class was virtually untested and was untestable in its previous state. I've added these two override points to allow tests to change the behaviour. I wanted to limit the scope of this change, so I didn't want to create any new interfaces or types at this point.